### PR TITLE
Deadlock fix: Do not listen on main queue

### PR DESCRIPTION
**Pull request by bwaresiak**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/414709148257752/1204381704707794/f
iOS PR: 
macOS PR: 
What kind of version bump will this require?: Patch

**Optional**:

Tech Design URL:
CC:

**Description**:

Fix potential deadlock situation.

**Steps to test this PR**:

Break in notification handler and validate the threads.

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/Bookmarks/MenuBookmarksViewModel.swift [hunk 1]
```diff
@@ -58,16 +60,25 @@ public class MenuBookmarksViewModel: MenuBookmarksInteracting {
...
+            NotificationCenter.default.removeObserver(observer)
+            self.observer = nil
+        }
+    }
+
     private func registerForChanges() {
-        NotificationCenter.default.addObserver(forName: NSManagedObjectContext.didSaveObjectsNotification,
-                                               object: nil,
-                                               queue: .main) { [weak self] notification in
+        observer = NotificationCenter.default.addObserver(forName: NSManagedObjectContext.didSaveObjectsNotification,
```
Comment (brindy) [inline Sources/Bookmarks/MenuBookmarksViewModel.swift:71]: Why not use the selector version now?  And then you also don't need all this extra code for unregistering, right?

https://developer.apple.com/documentation/foundation/notificationcenter/1415360-addobserver

Comment (bwaresiak) [reply Sources/Bookmarks/MenuBookmarksViewModel.swift:71]: I was thinking about that, and leaned towards not doing so:
- I don't like having to mark function as @objc just to use that API :)
- This shows good example of having to remove observer when using this queue based API - so I though about keeping that in the codebase, as it is easy to forget that some of the Notification Center API does handle unregistering on its own, and some not.

I don't have strong opinion tho, so if you care/feel it should be changed I can do this.

Comment (brindy) [reply Sources/Bookmarks/MenuBookmarksViewModel.swift:71]: Nah, I don't feel that strongly either :)

**Review by Bunn — APPROVED** [review]
LGTM.
Tested bookmarks on both iOS and macOS, as well as trying to reproduce the issue by closing/opening the app multiple times.

**Pull request merged**